core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/AsyncExecutionFailureExceptionReportingStage.java

```diff
@@ -33,12 +34,15 @@ public AsyncExecutionFailureExceptionReportingStage(RequestPipeline<SdkHttpFullR
 
     @Override
     public CompletableFuture<OutputT> execute(SdkHttpFullRequest input, RequestExecutionContext context) throws Exception {
-        return wrapped.execute(input, context).whenComplete((o, t) -> {
+        return wrapped.execute(input, context).handle((o, t) -> {
             if (t != null) {
                 Context.FailedExecution failedContext =
                         new DefaultFailedExecutionContext(context.executionContext().interceptorContext(), t);
+                //TODO: To obtrude or not to obtrude?
```

Comment:
What does obtrude mean?

Reply:
With this change, since this method gets called when an exception happens further up the pipeline, if the interceptor also throws, then we throw that exception instead of the original exception. Not sure what behavior we want

Reply:
Oh gotcha. I say we should favor the original exception if it's present.

Reply:
Sounds good, preserving the original exception